env/classroom_app/apps/serializers.py

```diff
@@ -104,4 +85,51 @@ class AllotedCourseSerializer(serializers.ModelSerializer):
     class Meta:
         model = AllotedCourse
         fields = '__all__'
-        
\ No newline at end of file
+
+
+class AssignmentSerializer(serializers.ModelSerializer):
+    assigned_by = TeacherSerializer()
+    course = CourseSerializer()
+    salary = serializers.SerializerMethodField()
```

Comment:
check if we can replace `serializermethod` field with Intergerfield using `source` keyword